rundown/resources/date.py

```diff
@@ -0,0 +1,27 @@
+from datetime import datetime
+
+from pydantic import BaseModel, validator
+import arrow
+
+from rundown.usercontext import context_timezone
+
+
+class Date(BaseModel):
+    date: str
+
+    @validator("date")
+    def make_timezone(cls, v):
+        """Format date returned from server.
+
+        The dates incorrectly always show UTC timezone, but the local time will be
+        correct with respect to timezone. So it is only necessary to truncate the date
+        string.
+        """
+        naive_dt_str = v.split("+")[0]
```

Comment:
It can. I noticed a bug in `/sports/<sport-id>/dates` route where when you change the offset the timezone doesn't change, but the time does. E.G:

```python
r = Rundown(<api-key>, timezone='America/Vancouver')
s = Rundown(<api-key>, timezone='UTC')
rd = r.dates_by_sport(6)
sd = s.dates_by_sport(6)

print(rd)
# output
{'dates': ['2021-04-05T16:00:00+0000',
  '2021-04-06T16:00:00+0000',
  '2021-04-07T14:00:00+0000',
  '2021-05-05T16:00:00+0000']}

print(sd)
# output
{'dates': ['2021-04-05T23:00:00+0000',
  '2021-04-06T00:00:00+0000',
  '2021-04-07T00:00:00+0000',
  '2021-04-08T01:00:00+0000',
  '2021-05-05T23:00:00+0000']}
```

So I handled this by truncating off the incorrect timezone part.

Reply:
Ideally we could fix it on the server side and then this code wouldn't be necessary

Reply:
>  I noticed a bug in /sports/<sport-id>/dates route where when you change the offset the timezone doesn't change, but the time does

Yeah, this endpoint is confusing. It is intentionally always returning the times of the 24-hour windows in UTC instead of the localized timezone, although now that you bring this up it make more sense to do what you're suggesting.